# Mini Project 4: Specialty Foods Inc. - SQL Exercises

## Introduction

**Business Context.** Specialty Foods Inc. is a food retailer focusing on the higher end of the market. You are a new member of the marketing team that was hired based on your data analytic skills. The company is interested in improving business results through more data-driven analysis and decision making. Traditionally the marketing department has launched campaigns to increase sales using qualitative analysis that has focused on previous experience and an understanding of the market. 
Given your data analytic skills, your manager has asked you to help the marketing team by  gathering insights into the type of customers the company has and the products they buy. You are also asked to review past campaigns and suggest improvements for future marketing campaigns. In addition to gaining a better understanding of the business your analysis should result in specific recommendations on how the company can improve business results.

**Objective:** The goal of this mini project 4 is to practice your SQL skills by querying and aggregating data. Additionally you will use Tableau to create professional data visualizations.

**Analysis / Data Analytics:** For your analysis your team should begin by answering the questions below. However these questions are aimed only to get you started and practice your skills. You should consider further analysis and determine what additional questions will help in both understanding the business and making recommendations to improve the business’s results. 

### Tools used in this project

SQL is a key tool used for data analysis. It is used for querying and accessing data, performing data cleaning, and analyzing data that is stored in databases. It performs some similar tasks that we have used Excel for but is a more powerful tool when manipulating larger datasets and provides some of the flexibility found in programming languages.

Tableau is a professional data visualization software that is widely used in industry. It is a great tool for storytelling with data. Once you are familiar with its functionality you should notice some similarities with Excel which will allow you to quickly get up to speed with the Tableau software. Use Tableau to create your charts similar to those in Excel for presenting your data insights and it can also be used for exploratory data analysis. 

## Overview of the data

The data for this case is contained in three separate tables which are extracts from the customer, sales, and marketing databases of the company and contain the information below:

**Customer table includes the following information:**
* **ID**: customer unique ID
* **Income**: customer’s yearly household income
* **Kids**: number of small children in the household
* **Teens**: number of teenagers in the household
* **Age**: age of customer
* **Divorced**: 1 if the person is divorced, 0 otherwise
* **Married**: 1 if the person is married, 0 otherwise
* **Single**: 1 if the person is single, 0 otherwise
* **Together**: 1 if the person is living with a partner, 0 otherwise
* **Widowed**: 1 if the person is widowed, 0 otherwise
* **Basic**: 1 if education is secondary level (high school), 0 otherwise
* **Graduate**: 1 if education is university level, 0 otherwise
* **Master**: 1 if education is masters level, 0 otherwise
* **PhD**: 1 if education is doctorate level, 0 otherwise
* **State**: US state of residency

**Sales table includes the following information:**
* **ID**: customer unique ID
* **Recency**: days since last purchase
* **Wines**: amount spent on wine
* **Fruits**: amount spent on fruit
* **Meats**: amount spent on meat
* **Seafood**: amount spent on seafood
* **Sweets**: amount spent on sweets
* **Premium**: amount spent on premium products
* **Regular**: amount spent on standard products
* **Deals**: number of purchases made with a discount
* **Web**: number of website purchases
* **Catalog**: number of catalog purchases
* **Store**: number of in-store purchases
* **Days**: number of days since last purchase
* **Visits**: number of website visits in past 3 months

**Marketing table includes the following information:**
* **ID**: customer unique ID
* **MC3**: 1 if customer made a purchase based on Campaign 3, otherwise 0
* **MC4**: 1 if customer made a purchase based on Campaign 4, otherwise 0
* **MC5**: 1 if customer made a purchase based on Campaign 5, otherwise 0
* **MC1**: 1 if customer made a purchase based on Campaign 1, otherwise 0
* **MC2**: 1 if customer made a purchase based on Campaign 2, otherwise 0
* **Complaint**: 1 if customer made a complaint in past year
* **Pilot**: 1 if customer made a purchase based on a recent pilot marketing campaign for a new product, otherwise 0
* **Enrollment**: date the customer enrolled with the company


### Your Task

In addition to brainstorming with your team members, you should discuss with your TAs and learn from other fellows to explore other techniques for using SQL and Tableau.

For your analysis your team should begin by answering the questions below. However these questions are aimed only to get you started and practice your skills. You should consider further analysis and determine what additional questions will help in both understanding the business and making recommendations to improve the business’s results. You can add cells to this jupyter notebook in order to run additional sql queries that can help you investigate the dataset and develop your analysis.

**Deliverables.**   

Place the code answers to the SQL questions in the Jupyter Notebook. Add as many additional code cells as you need in order to complete your analysis. Also include any written responses to the respective SQL questions, and the rationale for your additional SQL analyses in the Jupyter Notebook within a markdown (text) cell.  

Place the link(s) to your Tableau visualizations in a markdown cell at the end of the jupyter notebook.

### Getting Started

We will begin by loading a database containing Specialty Foods' customer, sales, and marketing data tables. These next two cells are not standard SQL, so just run them as is.

In [1]:
%FETCH https://amzn-dana.workspace-lite.correlation-one.com/case.sql_tableau_mini_project_fellow/files/data/specialtyfoods.db specialtyfoods

Start downloading from URL https://amzn-dana.workspace-lite.correlation-one.com/case.sql_tableau_mini_project_fellow/files/data/specialtyfoods.db
Finished downloading 249856 bytes from URL https://amzn-dana.workspace-lite.correlation-one.com/case.sql_tableau_mini_project_fellow/files/data/specialtyfoods.db
Writing downloaded data to file specialtyfoods
Finished writing file


In [2]:
%LOAD specialtyfoods RW

Using our new database query the tables to understand what type of data is included in each table.

Let's query the customer table in our database like this:

In [3]:
SELECT * FROM customer LIMIT 5;

+------+--------+------+-------+-----+----------+---------+--------+----------+-------+-------+----------+--------+-----+------------------+
| ID   | Income | Kids | Teens | Age | Divorced | Married | Single | Together | Widow | Basic | Graduate | Master | PhD | State            |
+------+--------+------+-------+-----+----------+---------+--------+----------+-------+-------+----------+--------+-----+------------------+
| 2873 | 213734 | 0.0  | 0.0   | 75  | 0        | 0       | 1      | 0        | 0     | 0     | 0        | 0      | 1   | State-California |
+------+--------+------+-------+-----+----------+---------+--------+----------+-------+-------+----------+--------+-----+------------------+
| 1629 | 205471 | 0.0  | 0.0   | 50  | 0        | 0       | 0      | 1        | 0     | 0     | 1        | 0      | 0   | State-Texas      |
+------+--------+------+-------+-----+----------+---------+--------+----------+-------+-------+----------+--------+-----+------------------+
| 1239 | 202692 | 0.0  | 0.0   | 46  | 1        | 0       | 0      | 0        | 0     | 0     | 1        | 0      | 0   | State-Texas      |
+------+--------+------+-------+-----+----------+---------+--------+----------+-------+-------+----------+--------+-----+------------------+
| 1191 | 202160 | 0.0  | 0.0   | 43  | 0        | 0       | 0      | 1        | 0     | 0     | 0        | 0      | 1   | State-Texas      |
+------+--------+------+-------+-----+----------+---------+--------+----------+-------+-------+----------+--------+-----+------------------+
| 1116 | 201970 | 0.0  | 0.0   | 37  | 0        | 0       | 1      | 0        | 0     | 0     | 1        | 0      | 0   | State-Florida    |
+------+--------+------+-------+-----+----------+---------+--------+----------+-------+-------+----------+--------+-----+------------------+

In [4]:
SELECT * FROM customer LIMIT 5;

+------+--------+------+-------+-----+----------+---------+--------+----------+-------+-------+----------+--------+-----+------------------+
| ID   | Income | Kids | Teens | Age | Divorced | Married | Single | Together | Widow | Basic | Graduate | Master | PhD | State            |
+------+--------+------+-------+-----+----------+---------+--------+----------+-------+-------+----------+--------+-----+------------------+
| 2873 | 213734 | 0.0  | 0.0   | 75  | 0        | 0       | 1      | 0        | 0     | 0     | 0        | 0      | 1   | State-California |
+------+--------+------+-------+-----+----------+---------+--------+----------+-------+-------+----------+--------+-----+------------------+
| 1629 | 205471 | 0.0  | 0.0   | 50  | 0        | 0       | 0      | 1        | 0     | 0     | 1        | 0      | 0   | State-Texas      |
+------+--------+------+-------+-----+----------+---------+--------+----------+-------+-------+----------+--------+-----+------------------+
| 1239 | 202692 | 0.0  | 0.0   | 46  | 1        | 0       | 0      | 0        | 0     | 0     | 1        | 0      | 0   | State-Texas      |
+------+--------+------+-------+-----+----------+---------+--------+----------+-------+-------+----------+--------+-----+------------------+
| 1191 | 202160 | 0.0  | 0.0   | 43  | 0        | 0       | 0      | 1        | 0     | 0     | 0        | 0      | 1   | State-Texas      |
+------+--------+------+-------+-----+----------+---------+--------+----------+-------+-------+----------+--------+-----+------------------+
| 1116 | 201970 | 0.0  | 0.0   | 37  | 0        | 0       | 1      | 0        | 0     | 0     | 1        | 0      | 0   | State-Florida    |
+------+--------+------+-------+-----+----------+---------+--------+----------+-------+-------+----------+--------+-----+------------------+

### Exercise 1

Simiarly, explore the marketing and sales tables.

In [5]:
SELECT * FROM marketing LIMIT 5;

+------+-----+-----+-----+-----+-----+-----------+-------+------------+
| ID   | MC3 | MC4 | MC5 | MC1 | MC2 | Complaint | Pilot | Enrollment |
+------+-----+-----+-----+-----+-----+-----------+-------+------------+
| 1188 | 0   | 1   | 0   | 0   | 0   | 0         | 0     | 12/3/2020  |
+------+-----+-----+-----+-----+-----+-----------+-------+------------+
| 1970 | 0   | 0   | 0   | 0   | 0   | 0         | 0     | 12/3/2020  |
+------+-----+-----+-----+-----+-----+-----------+-------+------------+
| 1043 | 0   | 0   | 0   | 0   | 0   | 0         | 0     | 12/2/2020  |
+------+-----+-----+-----+-----+-----+-----------+-------+------------+
| 1777 | 0   | 0   | 0   | 0   | 0   | 0         | 0     | 12/2/2020  |
+------+-----+-----+-----+-----+-----+-----------+-------+------------+
| 2787 | 0   | 0   | 0   | 1   | 0   | 0         | 0     | 12/2/2020  |
+------+-----+-----+-----+-----+-----+-----------+-------+------------+

In [6]:
SELECT * FROM sales LIMIT 5;

+------+---------+-------+--------+-------+---------+--------+---------+---------+-------+---------+---------+-------+--------+
| ID   | Recency | Wines | Fruits | Meats | Seafood | Sweets | Premium | Regular | Deals | Website | Catalog | Store | Visits |
+------+---------+-------+--------+-------+---------+--------+---------+---------+-------+---------+---------+-------+--------+
| 1428 | 99      | 0     | 36     | 18    | 42      | 36     | 72      | 60      | 1     | 1.0     | 0.0     | 3     | 8      |
+------+---------+-------+--------+-------+---------+--------+---------+---------+-------+---------+---------+-------+--------+
| 2152 | 99      | 0     | 36     | 18    | 42      | 36     | 72      | 60      | 1     | 1.0     | 0.0     | 3     | 8      |
+------+---------+-------+--------+-------+---------+--------+---------+---------+-------+---------+---------+-------+--------+
| 2014 | 99      | 21    | 42     | 55    | 13      | 34     | 68      | 97      | 1     | 1.0     | 0.0     | 3     | 8      |
+------+---------+-------+--------+-------+---------+--------+---------+---------+-------+---------+---------+-------+--------+
| 2660 | 99      | 81    | 0      | 27    | 0       | 3      | 13      | 98      | 1     | 1.0     | 0.0     | 3     | 5      |
+------+---------+-------+--------+-------+---------+--------+---------+---------+-------+---------+---------+-------+--------+
| 1196 | 99      | 175   | 0      | 23    | 0       | 0      | 13      | 184     | 1     | 2.0     | 0.0     | 3     | 6      |
+------+---------+-------+--------+-------+---------+--------+---------+---------+-------+---------+---------+-------+--------+

Write a query to understand your data in the sales table. 

This query calculates the average amount spent on each product category:

In [7]:
SELECT
    ROUND(AVG(Wines), 2) AS Average_Wines,
    ROUND(AVG(Fruits), 2) AS Average_Fruits,
    ROUND(AVG(Meats), 2) AS Average_Meats,
    ROUND(AVG(Seafood), 2) AS Average_Seafood,
    ROUND(AVG(Sweets), 2) AS Average_Sweets
FROM sales;

+---------------+----------------+---------------+-----------------+----------------+
| Average_Wines | Average_Fruits | Average_Meats | Average_Seafood | Average_Sweets |
+---------------+----------------+---------------+-----------------+----------------+
| 781.23        | 69.43          | 444.57        | 99.08           | 70.95          |
+---------------+----------------+---------------+-----------------+----------------+

This query calculates the total amount spent on each product category:

In [8]:
SELECT
    SUM(Wines) AS Total_Wines,
    SUM(Fruits) AS Total_Fruits,
    SUM(Meats) AS Total_Meats,
    SUM(Seafood) AS Total_Seafood,
    SUM(Sweets) AS Total_Sweets
FROM sales;

+-------------+--------------+-------------+---------------+--------------+
| Total_Wines | Total_Fruits | Total_Meats | Total_Seafood | Total_Sweets |
+-------------+--------------+-------------+---------------+--------------+
| 1734336     | 154139       | 986946      | 219967        | 157502       |
+-------------+--------------+-------------+---------------+--------------+

This query calculates the total number of unique customers:

In [9]:
SELECT COUNT(DISTINCT ID) AS Total_Customers
FROM sales;

+-----------------+
| Total_Customers |
+-----------------+
| 2220            |
+-----------------+

In [10]:
SELECT COUNT(DISTINCT ID) AS Total_Customers
FROM sales;

+-----------------+
| Total_Customers |
+-----------------+
| 2220            |
+-----------------+

This query calculates the average of days since the last purchase across all customers:

In [11]:
SELECT ROUND(AVG(Recency), 2) AS Average_Recency
FROM sales;

+-----------------+
| Average_Recency |
+-----------------+
| 49.03           |
+-----------------+

Write a query to understand your data in the marketing table.

This query calculates the count of customers who made purchases based on each campaign:

In [12]:
SELECT
    SUM(MC1) AS Purchased_in_Campaign_1,
    SUM(MC2) AS Purchased_in_Campaign_2,
    SUM(MC3) AS Purchased_in_Campaign_3,
    SUM(MC4) AS Purchased_in_Campaign_4,
    SUM(MC5) AS Purchased_in_Campaign_5
FROM Marketing;

+-------------------------+-------------------------+-------------------------+-------------------------+-------------------------+
| Purchased_in_Campaign_1 | Purchased_in_Campaign_2 | Purchased_in_Campaign_3 | Purchased_in_Campaign_4 | Purchased_in_Campaign_5 |
+-------------------------+-------------------------+-------------------------+-------------------------+-------------------------+
| 143                     | 30                      | 165                     | 165                     | 161                     |
+-------------------------+-------------------------+-------------------------+-------------------------+-------------------------+

This query calculates the count of customers who made purchases in any of the campaigns:

In [13]:
SELECT
    SUM(MC1 + MC2 + MC3 + MC4 + MC5) AS Customers_With_Purchases
FROM Marketing;

+--------------------------+
| Customers_With_Purchases |
+--------------------------+
| 664                      |
+--------------------------+

This query calculates the count of customers who did not make any purchases in the campaigns:

In [14]:
SELECT
    COUNT(*) - SUM(MC1 + MC2 + MC3 + MC4 + MC5) AS Customers_Without_Purchases
FROM Marketing;

+-----------------------------+
| Customers_Without_Purchases |
+-----------------------------+
| 1556                        |
+-----------------------------+

This query calculates the count of customers who made complaints:

In [15]:
SELECT
    SUM(Complaint) AS Customers_With_Complaints
FROM Marketing;

+---------------------------+
| Customers_With_Complaints |
+---------------------------+
| 20                        |
+---------------------------+

### Exercise 2

Query the database for the products that are purchased by customers based on marital status.

Specifically, join the customer and products tables and create a query for amount of wine (TotalWines) purchased by customers based on whether they are divorced or not.

Expected output:

Divorced	TotalWines

        0	1538225
        1	196111


In [16]:
SELECT customer.divorced AS Divorced, SUM(sales.wines) AS TOTALWines
FROM customer JOIN sales
ON customer.ID = sales.ID
GROUP BY customer.divorced;

+----------+------------+
| Divorced | TOTALWines |
+----------+------------+
| 0        | 1538225    |
+----------+------------+
| 1        | 196111     |
+----------+------------+

### Exercise 3

Query the database to determine what type of customers purchase which products. Can you describe what are the types of customers, e.g. what is the customer persona or segment.

Specifically, create a query to find the total amount spent on sweets (TotalSweets) for customers who have education above the university level and who are not single.

Expected output:

Master	PhD	TotalSweets

    0	1	17683
    1	0	15213



In [17]:
SELECT customer.master, customer.PhD, SUM(sales.Sweets) AS TotalSweets
FROM sales
JOIN customer ON sales.ID= customer.ID
WHERE ((customer.Master = 1 or customer.PhD = 1)) AND customer.single != 1
GROUP BY customer.Master, customer.PhD;

+--------+-----+-------------+
| Master | PhD | TotalSweets |
+--------+-----+-------------+
| 0      | 1   | 17683       |
+--------+-----+-------------+
| 1      | 0   | 15213       |
+--------+-----+-------------+

### Exercise 4

Query your database to discover which products bring in the most revenues for different customer segments.

Specifically, create a query to find the average age (AveAge) and average income (AveIncome) of customers from California along with their total sales for meats (TotalMeats) and seafood (TotalSeafood).

Expected output:

AveAge:  50.30810810810811

AveIncome:  152234.3135135135

TotalMeats:  147089

TotalSeafood:  40222


In [18]:
SELECT AVG(customer.age) AS AveAge,AVG(customer.income) AS AveIncome, SUM(sales.meats) AS TotalMeats, SUM(sales.seafood) AS TotalSeafood
FROM customer JOIN sales
ON customer.ID = sales.ID
WHERE customer.state = "State-California"

+------------------+------------------+------------+--------------+
| AveAge           | AveIncome        | TotalMeats | TotalSeafood |
+------------------+------------------+------------+--------------+
| 50.3081081081081 | 152234.313513514 | 147089     | 40222        |
+------------------+------------------+------------+--------------+

## SQL Bonus Question (Optional)

### Exercise 5

Create one query that outputs the total sales from premium products and the average income of customers for customers over the age of 50 who participated in marketing campaign 5. 

Hint: use two inner joins

Expected output:

TotalPremium:  12077

AveIncome:  181205.7972972973

In [19]:
SELECT SUM(sales.premium) AS TotalPremium, AVG(customer.income) AS AveIncome
FROM sales
JOIN customer ON sales.id = customer.id
JOIN marketing ON marketing.ID = customer.ID
WHERE customer.age > 50 
AND MC5 = 1;

+--------------+------------------+
| TotalPremium | AveIncome        |
+--------------+------------------+
| 12077        | 181205.797297297 |
+--------------+------------------+

### Your Analysis

This ends the directed part of the analysis. Be sure to continue your own analysis below, adding code and markdown cells as needed.

Q1:

1.1 Average Amount Spent on Each Product Category:

SELECT
    ROUND(AVG(Wines), 2) AS Average_Wines,
    ROUND(AVG(Fruits), 2) AS Average_Fruits,
    ROUND(AVG(Meats), 2) AS Average_Meats,
    ROUND(AVG(Seafood), 2) AS Average_Seafood,
    ROUND(AVG(Sweets), 2) AS Average_Sweets
FROM sales;

This query calculates and rounds the average spending on five different product categories: Wines, Fruits, Meats, Seafood, and Sweets.
It provides insights into the average customer spending patterns, with the highest average spending on Wines $781.23 and the lowest on Fruits $69.43.

1.2 Total Amount Spent on Each Product Category:

SELECT
    SUM(Wines) AS Total_Wines,
    SUM(Fruits) AS Total_Fruits,
    SUM(Meats) AS Total_Meats,
    SUM(Seafood) AS Total_Seafood,
    SUM(Sweets) AS Total_Sweets,
FROM sales;

This query calculates the total spending on the same product categories: Wines, Fruits, Meats, Seafood, Sweets.
It gives an overview of the overall revenue generated from each product category. Overall, it seems that wines, meats, and seafood are the most significant categories in terms of expenditure, while fruits and sweets have relatively lower spending. 

1.3 Total Number of Unique Customers:

SELECT COUNT(DISTINCT ID) AS Total_Customers
FROM sales;

This query counts the total number of unique customers by using the DISTINCT clause on the "ID" column.
There are 2,220 unique customers in the dataset.
 

1.4 Average Days Since the Last Purchase:

SELECT ROUND(AVG(Recency), 2) AS Average_Recency
FROM sales;

This query calculates the average number of days since the last purchase across all customers.
The average recency is approximately 49.03 days, which can be interpreted as the average gap between purchases for the customer base.

1.5 Count of Customers Who Made Purchases Based on Each Campaign:

SELECT
    SUM(MC1) AS Purchased_in_Campaign_1,
    SUM(MC2) AS Purchased_in_Campaign_2,
    SUM(MC3) AS Purchased_in_Campaign_3,
    SUM(MC4) AS Purchased_in_Campaign_4,
    SUM(MC5) AS Purchased_in_Campaign_5
FROM Marketing;


This query sums up the number of customers who made purchases in each marketing campaign.
It provides campaign-specific insights into customer engagement, with varying levels of participation in different campaigns.The campaign analysis demonstrates notable variations in their effectiveness. Campaigns 3 and 4 achieved significant success with 165 purchases each, suggesting well-executed strategies or enticing offers. Campaign 5 also showed promise with 161 purchases, indicating its potential. On the other hand, Campaign 1, with 143 purchases, performed moderately, while Campaign 2 lagged significantly behind with only 30 purchases, indicating the need for further evaluation and adjustment of its approach. 


1.6 Count of Customers Who Made Purchases in Any Campaign:

SELECT
    SUM(MC1 + MC2 + MC3 + MC4 + MC5) AS Customers_With_Purchases
FROM Marketing;

This query calculates the count of customers who made purchases in at least one campaign.
There are 664 customers who engaged with the marketing campaigns by making purchases.

1.7 Count of Customers Who Did Not Make Any Purchases in the Campaigns:

SELECT
    COUNT(*) - SUM(MC1 + MC2 + MC3 + MC4 + MC5) AS Customers_Without_Purchases
FROM Marketing;

This query calculates the count of customers who did not make any purchases in any of the campaigns.
There are 1,556 customers who did not participate in any of the campaigns by making purchases.

1.8 Count of Customers Who Made Complaints:

SELECT
    SUM(Complaint) AS Customers_With_Complaints
FROM Marketing;

This query sums up the number of customers who made complaints.
There are 20 customers who raised complaints.


Q2.

SELECT: specifies the columns to be included in the result set. It selects two columns:   customer.divorced AS Divorced: This column alias renames the "divorced" column from the "customer" table as "Divorced" in the result set and SUM (sales.wines) AS TOTALWines: This column calculates the total sum of wine purchases from the "sales" table and assigns the alias "TOTALWines" to this calculated value. 
FROM: specifies the tables involved in the query and their relationships. It joins two tables, customer: This table is a list of customers. sales: This table contains information about sales transactions.
JOIN: defined by ON customer.ID = sales.ID, indicating that the query matches records where the "ID" in the "customer" table matches the "ID" in the "sales" table. 
GROUP BY: used to group the results based on a specific column.In this case, the query groups the results by the "divorced" column from the "customer" table. This means that the query will produce separate rows for each unique value in the "divorced" column. 

Customers who are not divorced ("Divorced=0") have a significantly higher total wine purchase of 1,538,225 units compared to customers who are divorced ("Divorced=1"), who have a total wine purchase of 196,111 units. This suggests that customers who are not divorced may have a greater affinity for wine or make more frequent wine purchases.


Q3.SELECT :specifies the columns to be included in the result set. It selects three columns, customer.master: this column represents whether a customer has a master's degree, customer.phd: this column represents whether a customer has a PhD,SUM(sales.sweets) AS TotalSweets: This column calculates the total sum of "sweets" purchases from the "sales" table and assigns the alias "TotalSweets" to this calculated value
FROM: specifies the table involved It selects the "sales" table, which contains information about sales transactions
JOIN:used to join the "sales" table with the "customer" table based on a common column, which is the "ID" column in both tables. WHERE :filters the rows from the joined tables. It filters rows where the "customer.single" column equals 1. This represents customers who are single.
GROUP BY:used to group the results based on the "customer.master" and "customer.phd" columns from the "customer" table. this means that the query will produce separate rows for each unique combination of values in these two columns. 

Customers with a Master's degree have a total sweets purchase ("TotalSweets") of 17,683 units and customers with a PhD have a total sweets purchase of 15,213 units. We can notice Master's degrees, shows a higher total sweets purchase and customers with PhDs also purchases sweets, although their total purchases are slightly lower than the Master's degree segment.

    
Q4.SELECT: specifies the columns to be included in the result set. It selects three columns with calculated values: 
    AVG(customer.age) AS AveAge: This calculates the average age of customers in California and assigns the alias "AveAge" to this value. 
    AVG(customer.income) AS AveIncome: This calculates the average income of customers in California and assigns the alias "AveIncome" to this value. SUM(sales.meats) AS TotalMeat: This calculates the total sum of meat purchases from the "sales" table for customers in California and assigns the alias "TotalMeat" to this calculated value. 
FROM:specifies the tables involved in the query. It selects the "customer" table, which contains customer information, and the "sales" table, which contains sales transaction data. 
JOIN: used to join the "customer" and "sales" tables based on a common column, which is the "ID" column in both tables. This links customer data to their sales transactions. 
WHERE: filters the rows from the joined tables. It filters rows where the "customer.state" column equals "State-California." This ensures that only data related to customers from California is considered in the calculations. This SQL query calculates and presents an analysis of customers in California. It provides the average age and average income of customers in the state, along with the total sum of meat purchases made by these customers. The result set will contain three columns: "AveAge" (average age), "AveIncome" (average income), and "TotalMeat" (total meat purchases) for customers in California. 

On average, customers are around 50 years old and have a relatively high average income of approximately $152,234.31, meat products are quite popular among California customers, with total sales reaching 147,089 units, while seafood sales also contribute significantly, totaling 40,222 units.

Q5.SELECT: specifies the columns to be included in the result set. It selects two columns with calculated values: SUM(sales.premium) AS TotalPremium: This calculates the total sum of "premium" values from the "sales" table and assigns the alias "TotalPremium" to this calculated value. AVG(customer.income) AS AveIncome: This calculates the average income of customers and assigns the alias "AveIncome" to this value.
FROM: specifies the tables involved in the query. It selects the "sales" table, the "customer" table, and the "marketing" table. 
JOIN: Two JOIN clauses are used to join the tables based on common columns: JOIN customer ON sales.id = customer.id: This joins the "sales" and "customer" tables based on the "id" column, are linking sales transactions to customer information. JOIN marketing ON marketing.ID = customer.ID: This joins the "marketing" and "customer" tables based on the "ID" column, likely linking marketing data to customer information. 
WHERE: filters the rows from the joined tables based on specified conditions.  customer.age > 50: This condition filters customers with an age greater than 50. MC5 = 1: This condition filters records where the "MC5" column equals 1.


The results reveal that customers over the age of 50 who participated in marketing campaign 5 generated a total of 12,077 units in sales from premium products, with an impressive average income of approximately $181,205.80. These insights suggest the effectiveness of targeting specific age groups within marketing campaigns and emphasize the significance of premium product sales within this demographic.

## Final Analysis Specialty Foods Inc.

For Specialty Foods Inc., the data analysis provides valuable insights into customer spending patterns, campaign effectiveness, and customer demographics. The company observes that wines are the highest-selling category, with an average spending of $781.23, while fruits have the lowest average spending at $69.43. Meats and seafood are significant categories for expenditure, whereas fruits and sweets have relatively lower spending levels. With a customer base of 2,220 unique customers, Specialty Foods Inc. can leverage this information to tailor marketing strategies.

The analysis of marketing campaigns reveals varying success rates. Campaigns 3 and 4 are the most successful, each generating 165 purchases, suggesting well-executed strategies. Campaign 5 also shows promise with 161 purchases, while Campaign 1 performs moderately, and Campaign 2 requires further evaluation and adjustment.

The average recency of approximately 49.03 days signifies the average gap between customer purchases. Of the 2,220 customers, 664 engaged with the marketing campaigns by making purchases, while 1,556 did not participate in any of the campaigns. Additionally, 20 customers raised complaints, indicating a need for customer service improvements.

Demographically, customers who are not divorced tend to have significantly higher wine purchases compared to divorced customers, implying different preferences. Customers with Master's degrees and PhDs show interest in sweets, with Master's degree holders making slightly higher total purchases.

On average, customers are around 50 years old with a relatively high average income of approximately $152,234.31. Meat products are popular among California customers, with total sales of 147,089 units, while seafood also contributes significantly at 40,222 units.

Furthermore, customers over the age of 50 participating in marketing campaign 5 generated substantial sales of premium products, emphasizing the effectiveness of targeting specific age groups within campaigns. These insights provide Specialty Foods Inc. with valuable information for refining marketing strategies and product offerings to maximize profitability and customer satisfaction.



Tableu Link: https://public.tableau.com/views/MiniProject4_16945282964260/SpecialtyFoodsInc_DashboardIncomeOverview?:language=en-US&publish=yes&:display_count=n&:origin=viz_share_link